drivers/usb/device/usb_dc_sam_usbc.c

```diff
@@ -0,0 +1,1160 @@
+		}
+	}
+
+	return 0;
+}
+
+static void usb_dc_ep_enable_interrupts(uint8_t ep_idx)
+{
+	if (ep_idx == 0U) {
+		/* Control endpoint: enable SETUP */
+		regs->UECONSET[ep_idx] = USBC_UECON0SET_RXSTPES;
+
+	} else if (regs->UECFG[ep_idx] & USBC_UECFG0_EPDIR_IN) {
+		/* TX - IN direction: acknowledge FIFO empty interrupt */
+		regs->UESTACLR[ep_idx] = USBC_UESTA0CLR_TXINIC;
+		regs->UECONSET[ep_idx] = USBC_UECON0SET_TXINES;
+
+	} else {
+		/* RX - OUT direction */
+		regs->UECONSET[ep_idx] = USBC_UECON0SET_RXOUTES;
+	}
+}
+
+static void usb_dc_ep_isr_sta(uint8_t ep_idx)
+{
+	uint32_t sr = regs->UESTA[ep_idx];
+
+#if defined(CONFIG_USB_DRIVER_LOG_LEVEL_DBG)
+	if (regs->UESTA[ep_idx] != dev_ep_sta_dbg[0][ep_idx]) {
+		dev_ep_sta_dbg[0][ep_idx] = regs->UESTA[ep_idx];
+		dev_ep_sta_dbg[1][ep_idx] = 0;
+
+		LOG_DBG("ISR[%d] CON=%08x INT=%08x INTE=%08x "
+			"ECON=%08x ESTA=%08x%s", ep_idx,
+			regs->UDCON, regs->UDINT, regs->UDINTE,
+			regs->UECON[ep_idx], regs->UESTA[ep_idx],
+			((sr & USBC_UESTA0_RXSTPI) ? " STP" : ""));
+	} else if (dev_ep_sta_dbg[0][ep_idx] != dev_ep_sta_dbg[1][ep_idx]) {
+		dev_ep_sta_dbg[1][ep_idx] = dev_ep_sta_dbg[0][ep_idx];
+
+		LOG_DBG("ISR[%d] CON=%08x INT=%08x INTE=%08x "
+			"ECON=%08x ESTA=%08x LOOP", ep_idx,
+			regs->UDCON, regs->UDINT, regs->UDINTE,
+			regs->UECON[ep_idx], regs->UESTA[ep_idx]);
+	}
+#endif
+	if (sr & USBC_UESTA0_RAMACERI) {
+		regs->UESTACLR[ep_idx] = USBC_UESTA0CLR_RAMACERIC;
+
+		LOG_ERR("ISR: EP%d RAM Access Error", ep_idx);
+	}
+}
+
+static void usb_dc_ep0_isr(void)
+{
+	uint32_t sr = regs->UESTA[0];
+	uint32_t dev_ctrl = regs->UDCON;
+
+	usb_dc_ep_isr_sta(0);
+
+	regs->UECONCLR[0] = USBC_UECON0CLR_NAKINEC;
+	regs->UECONCLR[0] = USBC_UECON0CLR_NAKOUTEC;
+
+	if (sr & USBC_UESTA0_RXSTPI) {
+
+		regs->UESTACLR[0] = USBC_UESTA0CLR_NAKINIC;
+		regs->UESTACLR[0] = USBC_UESTA0CLR_NAKOUTIC;
+
+		if (sr & USBC_UESTA0_CTRLDIR) {
+			/** IN Package */
+			/** Nothing to do */
+		} else {
+			/** OUT Package */
+			regs->UECONSET[0] = USBC_UECON0SET_RXOUTES;
+		}
+
+		/* SETUP data received */
+		dev_data.ep_data[0].cb_out(USB_EP_DIR_OUT, USB_DC_EP_SETUP);
+
+		return;
+	}
+
+	if (sr & USBC_UESTA0_RXOUTI) {
+
+		/* OUT (to device) data received */
+		dev_data.ep_data[0].cb_out(USB_EP_DIR_OUT, USB_DC_EP_DATA_OUT);
+	}
+
+	if ((sr & USBC_UESTA0_TXINI) &&
+	    (regs->UECON[0] & USBC_UECON0_TXINE)) {
+
+		regs->UECONCLR[0] = USBC_UECON0CLR_TXINEC;
+
+		if (sr & USBC_UESTA0_CTRLDIR) {
+			/** Finish Control Write State */
+			return;
+		}
+
+		/* IN (to host) transmit complete */
+		dev_data.ep_data[0].cb_in(USB_EP_DIR_IN, USB_DC_EP_DATA_IN);
+
+		if (!(dev_ctrl & USBC_UDCON_ADDEN) &&
+		    (dev_ctrl & USBC_UDCON_UADD_Msk) != 0U) {
+			/* Commit the pending address update.  This must be
+			 * done after the ack to the host completes else the
+			 * ack will get dropped.
+			 */
+			regs->UDCON |= USBC_UDCON_ADDEN;
+		}
+	}
+
+	if (sr & USBC_UESTA0_NAKOUTI) {
+
+		/** Start Control Read State */
```

Comment:
you can remove empty lines above/below, same for `/** Start Control Write State */`